draft-ietf-tls-tls13.md

```diff
@@ -2942,22 +2943,47 @@ Any records following a 1-RTT Finished message MUST be encrypted under the
 application traffic key. In particular, this includes any alerts sent by the
 server in response to client Certificate and CertificateVerify messages.
 
-## Post-Handshake Messages
+## Post-Handshake Messages {#post-handshake}
 
 TLS also allows other messages to be sent after the main handshake.
 These messages use a handshake content type and are encrypted under the application
 traffic key.
 
+The post_handshake TLS extension advertises support for post-handshake message types.
+
+~~~~
+    enum {
+        new_session_ticket(0),
+        key_update(1),
+        client_auth_request(2),
+        (255)
+    } MessageType;
+
+    struct {
+        MessageType message_types<0..2^8-1>;
+    } PostHandshake;
+~~~~
+
+The extension data for the post_handshake extension is a PostHandshake structure,
+which includes at least one MessageType. The client includes a
+post_handshake extension containing every type of post-handshake message
+it supports in its ClientHello. If the server supports any post-handshake
+message types in common with the client, it replies with an
+EncryptedExtensions containing a post_handshake extension containing a list of
```

Comment:
The decision on whether to send EncryptedExtensions is not just based on PostHandshake; "it includes a post_handshake extension in the EncryptedExtensions it returns to the client.  The returned post_hanshake extension contains" might be a better wording.